src/main.rs

```diff
@@ -255,6 +258,24 @@ mod app {
         telemetry_task::spawn_after(((telemetry_period * 1000.0) as u64).millis()).unwrap();
     }
 
+    // Higher priority than telemetry but lower than adc data readout.
+    // 8 capacity to allow for max. 8 conversions to be queued.
+    #[task(priority = 2, shared=[channel_temperature, telemetry], capacity = 8)]
+    fn convert_adc_code(
+        mut c: convert_adc_code::Context,
+        input_ch: InputChannel,
+        adc_code: AdcCode,
+    ) {
+        // convert ADC code to °C and store in channel_temperature array and telemetry
+        c.shared.channel_temperature.lock(|channel_temperature| {
```

Comment:
Just a note:

There is a slightly nicer syntax to lock multiple resources in a tuple. That prevents the rightward drift.
But fine here.